libportal/remote.c

```diff
@@ -516,7 +558,11 @@ xdp_portal_create_remote_desktop_session (XdpPortal *portal,
   call->multiple = (flags & XDP_REMOTE_DESKTOP_FLAG_MULTIPLE) != 0;
   call->task = g_task_new (portal, cancellable, callback, data);
 
-  create_session (call);
+  if (portal->screencast_interface_version == 0 ||
+      portal->remote_desktop_interface_version == 0)
+    get_screencast_interface_version (call);
```

Comment:
`get_screencast_interface_version()` does not check if the screencast interface version was already fetched. If the operation is cancelled between `get_screencast_interface_version()` and `get_remote_desktop_interface_version()`, we'll end up with `portal->screencast_interface_version = 6` and ``portal->remote_desktop_interface_version = 0`, and this will fetch the interface version again unnecessarily.

It's not a big deal, but I'd rather make this code tighter. I think the simplest approach is adding a new function as follows:

```c
static void
get_portal_interface_versions (CreateCall *call)
{
  g_assert (call != NULL);
  g_assert (call->portal != NULL);

  if (call->portal->screencast_interface_version == 0)
    get_screencast_interface_version (call);
  else if (call->portal->remote_desktop_interface_version == 0)
    get_remote_desktop_interface_version (call);
  else
    create_session (call);
}
```

Reply:
done, thanks. And all the other style changes are done as requested too